### PR TITLE
More fixes

### LINKED ISSUE #148: celery.py should read DJANGO_SETTINGS_MODULE the same way manage.py does
Currently celery.py looks only for `os.environ`

### LINKED ISSUE #172: Unable to run npm build when project has custom assets (fonts, images, etc.) with relative paths
Although working in development, if I have an asset defined with a relative path, such as:

```css
background-image: url('./assets/images/placeholder.jpg');
```

I get this error when running `npm run build`:

```
Module build failed: ModuleNotFoundError: Module not found: Error: Can't resolve './assets/images/placeholder.jpg' in '/Users/---/project/node_modules/bootstrap-loader'
```

### LINKED ISSUE #212: Add circular-dependency-plugin to local webpack
**Is your feature request related to a problem? Please describe.**
It's easy to have circular imports in JS without anyone notice. Although this is not a problem in most cases it becomes more explicit when you are using Styled Components since they require the use of imported components at the root level of files.

**Describe the solution you'd like**
Add [circular-dependency-plugin](https://github.com/aackerman/circular-dependency-plugin) by default in the local webpack config

### LINKED ISSUE #367: Add default config to prevent session cookie issue
Defaults for below prevent the CSRF or session cookie from being sent in cross-site requests. Options are 'Lax' (default), 'Strict', and None. Those must be None due to a bug in Safari 12: 
- https://bugs.webkit.org/show_bug.cgi?id=188165
- https://www.calazan.com/django-21-samesite-cookie-issue-with-safari-12/
- https://github.com/axios/axios/issues/1661#issuecomment-463468120

```python
CSRF_COOKIE_SAMESITE = None
SESSION_COOKIE_SAMESITE = None
```

### LINKED ISSUE #368: Add template cached loader to settings
Caching templates

https://docs.djangoproject.com/en/2.2/ref/templates/api/#django.template.loaders.cached.Loader

```python
TEMPLATES[0]['OPTIONS']['loaders'] = [
    ('django.template.loaders.cached.Loader', [
        'django.template.loaders.filesystem.Loader',
        'django.template.loaders.app_directories.Loader',
    ]),
]
```

### LINKED ISSUE #498: ImportError: cannot import name 'celery_app'
**Description**

When I wanted to run a celery task, I got an error:

```
from config import celery_app

@celery_app.task(name="task_name")
def my_task():
    return

```

```
ImportError: cannot import name 'celery_app' from 'config' (/Users/matthieuchabert/Documents/workspace/stockpro/code/transfer/backend/config/__init__.py)
```

**Expected behavior**
According to the [Celery documentation](https://docs.celeryproject.org/en/stable/django/first-steps-with-django.html), we need to add the code below to the file `{{project_name}}/__init__.py`

```
from __future__ import absolute_import, unicode_literals

# This will make sure the app is always imported when
# Django starts so that shared_task will use this app.
from .celery import app as celery_app

__all__ = ('celery_app',)
```

I added it, and now it's working properly.

### LINKED ISSUE #511: heroku-redis version in app.json throws error
https://github.com/vintasoftware/django-react-boilerplate/blob/c4a203255ce4374bfa2346123a487adb29d877f4/app.json#L49-L51

image

### PR DESCRIPTION
**Pull request by rvlb**
## Description

Closes #148 (Celery getting current settings)
Closes #172 (frontend assets serving on production)
Closes #212 (Circular Dependency)
>CircularDependencyPlugin can be tested by adding `import App from '../App';` on `Home.js`

Closes #367 (Safari 12 bug)
Closes #368 (Add template cached loader to settings)
Closes #498 (Celery import)
Closes #511 (Heroku Deployment)

Partial fix for #115 (checklist for rendering the django logos)

Deployed example app https://foo-proj.herokuapp.com/

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires documentation updates.
- [ ] I have updated the documentation accordingly.
- [x] My change requires dependencies updates.
- [ ] I have updated the dependencies accordingly.

### CODE REVIEW

File: backend/project_name/__init__.py [hunk 1]
```diff
@@ -0,0 +1,3 @@
+from __future__ import absolute_import
```
Comment (fjsj) [inline backend/project_name/__init__.py:1]: I don't think we need __future__ as we're not Python 2 compatible. Please remove this elsewhere.

File: backend/project_name/settings/base.py [hunk 2]
```diff
@@ -56,7 +56,6 @@ def base_dir_join(*args):
     {
         "BACKEND": "django.template.backends.django.DjangoTemplates",
         "DIRS": [base_dir_join("templates")],
-        "APP_DIRS": True,
```
Comment (fjsj) [inline backend/project_name/settings/base.py:59]: Please test if this breaks apps with templates inside of them. Third-party install-able apps.

Comment (fjsj) [reply backend/project_name/settings/base.py:59]: Like this one: https://github.com/django-crispy-forms/django-crispy-forms

Comment (rvlb) [reply backend/project_name/settings/base.py:59]: I've tested rendering a Crispy Form using settings without `APP_DIRS` and it worked properly, but maybe this specific lib  supports that configuration by default.

File: app.json [hunk 3]
```diff
@@ -40,14 +40,14 @@
...
-        "version": "9.5"
+        "version": "9.6"
       },
       "as": "DATABASE"
     },
     {
       "plan": "heroku-redis:hobby-dev",
       "options": {
-        "version": "3.2"
+        "version": "6"
```
Comment (rvlb) [inline app.json:50]: These were the closest available versions for hobby-dev.

File: backend/project_name/settings/base.py (hunk 2 above)
Comment (fjsj) [reply backend/project_name/settings/base.py:59]: OK!

File: app.json (hunk 3 above)
Comment (fjsj) [reply app.json:50]: you can update to the last version, all plugins.

Comment (fjsj) [reply app.json:50]: Including Postgres.

**Review by fjsj — APPROVED** [review]
LGTM, thanks!

**Pull request merged**